### ISSUE TITLE
Pygame sprite class false error

### ISSUE THREAD

**Issue by PlayerUmut** [microsoft/pyright#10221]
I was watching a tutorial on coding a pygame game, there appears to be an error in the code. When i run it, it works fine. What can be the problem? Its at the last line, saying;
"center" is not a known attribute of "None"

```python
import pygame
class Player(pygame.sprite.Sprite):
    def __init__(self, groups):
        super().__init__(groups)
        self.image = pygame.image.load(join("images", "player.png")).convert_alpha()
        self.rect = self.image.get_frect(center=(WINDOW_WIDTH / 2, WINDOW_HEIGHT / 2))
        self.direction = pygame.math.Vector2()
        self.speed = 300
    def update(self, dt):
        keys = pygame.key.get_pressed()
        recent_keys = pygame.key.get_just_pressed()
        self.direction.x = int((keys[pygame.K_RIGHT] or keys[pygame.K_d]) - (keys[pygame.K_LEFT] or keys[pygame.K_a]))
        self.direction.y = int((keys[pygame.K_DOWN] or keys[pygame.K_s]) - (keys[pygame.K_UP] or keys[pygame.K_w]))
        self.direction = self.direction.normalize() if self.direction else self.direction
        self.rect.center += self.direction * self.speed * dt

```

pygame-ce 2.5.3 (SDL 2.30.12, Python 3.12.3)

Lazyvim (Neovim fork) V0.11.0,  MasonV1.11.0

**Comment by erictraut** [comment]
Pyright is a static type checker. It validates your code based on type information provided by the type information supplied by the library. Static type checkers don't run the code. Instead, they consider all possible paths through the code and report potential bugs.

I'll note that the code sample you've provided above is incomplete. It references several symbols that are not defined (`join`, `WINDOW_WIDTH`, etc.). When supplying code samples in bug reports, please make sure your code is complete and self-contained.

Now, let's take a look at why pyright is reporting an error here. The `rect` property in the `Sprite` class is declared as follows in the pygame library:

```python
class Sprite(_SupportsSprite):
    ...
    @property
    def rect(self) -> Optional[Union[FRect, Rect]]: ...
```
The type annotation `Optional[Union[FRect, Rect]]` means that this type might be `None` or an `FRect` or a `Rect`. If it is a `None` object, your program will crash when it attempts to access the `center` attribute because `None` has no such attribute. In other words, pyright is telling you that you have a potential bug in your code.

For basic information how static type checkers work, refer to [this documentation](https://microsoft.github.io/pyright/#/type-concepts).

If you suspect that the type information provided by the `pygame` library is incorrect, contact the maintainers of this library.

Closing the issue because pyright is working correctly here.
